kolibri/core/assets/src/heartbeat.js

```diff
@@ -45,15 +45,25 @@ class HeartBeat {
     return this.timerId;
   }
   beat() {
+    const sessionModel = this.kolibri.resources.SessionResource.getModel('current');
+    const userId = sessionModel.attributes.user_id;
     if (this.active) {
       logging.debug('There was activity, polling session endpoint!');
-      this.kolibri.resources.SessionResource.getModel('current').fetch({}, true).catch((error) => {
+      this.kolibri.resources.SessionResource.getModel('current').fetch({}, true).then(session => {
+        if (session.user_id !== userId) {
+          window.location.reload();
+        }
+      }).catch((error) => {
         logging.error('Periodic server polling failed, with error: ', error);
       });
       this.setActivityListeners();
     } else {
       logging.debug('No user activity, polling keepalive endpoint to keep session active');
-      this.kolibri.resources.KeepAliveResource.getModel('now').fetch({}, true).catch((error) => {
+      this.kolibri.resources.KeepAliveResource.getModel('now').fetch({}, true).then(session => {
+        if (session.user_id !== userId) {
+          window.location.reload();
+        }
```

Comment:
it's slightly bothering me that we're re-using the same heartbeat for both progress logging and keep-alive:

* the branching code path based on `this.active` seems likely to introduce bugs, since we're duplicating logic in both of them
* the two kinds of polling are for different functional purposes, and we may want them to run at different frequencies
* the performance benefits to running them on the same timer might not be that big (?)

Reply:
Could do keep alive unconditionally, and only do the user_id check in there.

Reply:
yeah, that sounds cleaner to me assuming there are no server load concerns

Reply:
I was going to implement that, but then changed my mind.

Instead, I've destroyed the keepalive endpoint, and just allowed an active flag to be set to 'false' on the session endpoint.

So, we always call the same endpoint, and there's no duplication of logic.

Reply:
I see the concern about the separation of concerns, but until we identify that we want to separate them, I think this is fine to keep them together.